diff -ruN src/SuiteSparse/COLAMD/Makefile src/SuiteSparse-patched/COLAMD/Makefile
--- src/SuiteSparse/COLAMD/Makefile	1969-12-31 19:00:00
+++ src/SuiteSparse-patched/COLAMD/Makefile	2024-01-12 11:54:13
@@ -0,0 +1,18 @@
+sources = Source/colamd.c Source/colamd_l.c Source/colamd_version.c
+objects = $(sources:.c=.o)
+archive = COLAMD.a
+
+PKG_CPPFLAGS = -I./Include -I../SuiteSparse_config
+
+all : $(archive)
+
+$(archive) : $(objects)
+	rm -f $@
+	$(AR) -cr $@ $(objects)
+	$(RANLIB) $@
+
+.c.o :
+	$(CC) $(PKG_CPPFLAGS) $(CPPFLAGS) $(CFLAGS) -c $< -o $*.o
+
+clean :
+	@rm -f $(objects) $(archive)
