diff -ruN src/SuiteSparse/SuiteSparse_config/Makefile src/SuiteSparse-patched/SuiteSparse_config/Makefile
--- src/SuiteSparse/SuiteSparse_config/Makefile	1969-12-31 19:00:00
+++ src/SuiteSparse-patched/SuiteSparse_config/Makefile	2024-03-16 01:09:07
@@ -0,0 +1,16 @@
+sources = SuiteSparse_config.c
+objects = $(sources:.c=.o)
+archive = SuiteSparse_config.a
+
+all : $(archive)
+
+$(archive) : $(objects)
+	rm -f $@
+	$(AR) -cr $@ $(objects)
+	$(RANLIB) $@
+
+.c.o :
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $*.o
+
+clean :
+	@rm -f $(objects) $(archive)
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c	2024-01-24 19:45:24
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c	2024-03-16 01:09:07
@@ -72,8 +72,8 @@
             /* MATLAB mexFunction: */
             mexPrintf,
         #else
-            /* standard ANSI C: */
-            printf,
+#include <R_ext/Print.h>
+            Rprintf,
         #endif
     #else
         /* printf is disabled */
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h	2024-01-24 19:45:24
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h	2024-03-16 01:12:06
@@ -41,6 +41,33 @@
 #include <stdarg.h>
 #include <ctype.h>
 
+#ifndef INT32_MAX
+# define int32_t   long
+# define INT32_MIN LONG_MIN
+# define INT32_MAX LONG_MAX
+# define PRId32    "ld"
+# define SCNd32    "ld"
+#endif
+#ifndef INT64_MAX
+# define int64_t   long long
+# define INT64_MIN LLONG_MIN
+# define INT64_MAX LLONG_MAX
+# define PRId64    "lld"
+# define SCNd64    "lld"
+#endif
+#ifndef UINT32_MAX
+# define uint32_t   unsigned long
+# define UINT32_MAX ULONG_MAX
+# define PRIu32     "lu"
+# define SCNu32     "lu"
+#endif
+#ifndef UINT64_MAX
+# define uint64_t   unsigned long long
+# define UINT64_MAX ULLONG_MAX
+# define PRIu64     "llu"
+# define SCNu64     "llu"
+#endif
+
 //------------------------------------------------------------------------------
 // SuiteSparse_long is now int64_t in SuiteSparse v6.0.0 and later
 //------------------------------------------------------------------------------
@@ -677,6 +704,19 @@
 
 #if defined ( SUITESPARSE_BLAS_DEFINITIONS )
 
+#ifndef USE_FC_LEN_T
+# define USE_FC_LEN_T
+#endif
+#include <Rconfig.h>
+
+#ifdef FC_LEN_T
+# define FCLEN , FC_LEN_T
+# define FCONE , (FC_LEN_T) 1
+#else
+# define FCLEN
+# define FCONE
+#endif
+
 //------------------------------------------------------------------------------
 // gemv: Y = alpha*A*x + beta*Y
 //------------------------------------------------------------------------------
@@ -696,7 +736,7 @@
     // input/output:
     double *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -709,7 +749,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -728,7 +768,7 @@
     // input/output:
     float *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -741,7 +781,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -760,7 +800,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -773,7 +813,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -792,7 +832,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -805,7 +845,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -825,7 +865,7 @@
     // input/output:
     double *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -836,7 +876,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -852,7 +892,7 @@
     // input/output:
     float *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -863,7 +903,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -879,7 +919,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -890,7 +930,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -906,7 +946,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -917,7 +957,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -940,7 +980,7 @@
     // input/output:
     double *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -952,7 +992,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -971,7 +1011,7 @@
     // input/output:
     float *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -983,7 +1023,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1002,7 +1042,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1014,7 +1054,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1033,7 +1073,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1045,7 +1085,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1070,7 +1110,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1086,7 +1126,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_DGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1107,7 +1147,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1123,7 +1163,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_SGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1144,7 +1184,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1160,7 +1200,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1181,7 +1221,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1197,7 +1237,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_CGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1219,7 +1259,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dsyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1231,7 +1271,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1249,7 +1289,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ssyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1261,7 +1301,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1279,7 +1319,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1291,7 +1331,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1309,7 +1349,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1321,7 +1361,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1339,7 +1379,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1351,7 +1391,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_DPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1366,7 +1406,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_spotrf(uplo,n,A,lda,info,ok)                       \
@@ -1378,7 +1418,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_SPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1393,7 +1433,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1405,7 +1445,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_ZPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1420,7 +1460,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_cpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1432,7 +1472,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_CPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1666,7 +1706,7 @@
     // output:
     double *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1678,7 +1718,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1695,7 +1735,7 @@
     // output:
     float *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1707,7 +1747,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1724,7 +1764,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1736,7 +1776,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1753,7 +1793,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1765,7 +1805,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1794,7 +1834,7 @@
     // workspace:
     double *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1811,7 +1851,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1836,7 +1876,7 @@
     // workspace:
     float *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1853,7 +1893,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1878,7 +1918,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1895,7 +1935,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1920,7 +1960,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1937,7 +1977,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -2139,7 +2179,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    double *Work
+    double *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2151,7 +2191,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2169,7 +2209,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    float *Work
+    float *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2181,7 +2221,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2199,7 +2239,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2211,7 +2251,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2229,7 +2269,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2241,7 +2281,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
